core/base/mixed_precision_types.hpp

```diff
@@ -0,0 +1,108 @@
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#ifndef GKO_CORE_BASE_MIXED_PRECISION_TYPES_HPP_
+#define GKO_CORE_BASE_MIXED_PRECISION_TYPES_HPP_
+
+
+#include <ginkgo/config.hpp>
+#include <ginkgo/core/base/types.hpp>
+
+
+#ifdef GINKGO_MIXED_PRECISION
+#define GKO_INSTANTIATE_FOR_EACH_MIXED_VALUE_AND_INDEX_TYPE(_macro) \
+    template _macro(float, float, float, int32);                    \
+    template _macro(float, float, double, int32);                   \
+    template _macro(float, double, float, int32);                   \
+    template _macro(float, double, double, int32);                  \
+    template _macro(double, float, float, int32);                   \
+    template _macro(double, float, double, int32);                  \
+    template _macro(double, double, float, int32);                  \
+    template _macro(double, double, double, int32);                 \
+    template _macro(std::complex<float>, std::complex<float>,       \
+                    std::complex<float>, int32);                    \
+    template _macro(std::complex<float>, std::complex<float>,       \
+                    std::complex<double>, int32);                   \
+    template _macro(std::complex<float>, std::complex<double>,      \
+                    std::complex<float>, int32);                    \
+    template _macro(std::complex<float>, std::complex<double>,      \
+                    std::complex<double>, int32);                   \
+    template _macro(std::complex<double>, std::complex<float>,      \
+                    std::complex<float>, int32);                    \
+    template _macro(std::complex<double>, std::complex<float>,      \
+                    std::complex<double>, int32);                   \
+    template _macro(std::complex<double>, std::complex<double>,     \
+                    std::complex<float>, int32);                    \
+    template _macro(std::complex<double>, std::complex<double>,     \
+                    std::complex<double>, int32);                   \
+    template _macro(float, float, float, int64);                    \
+    template _macro(float, float, double, int64);                   \
+    template _macro(float, double, float, int64);                   \
+    template _macro(float, double, double, int64);                  \
+    template _macro(double, float, float, int64);                   \
```

Comment:
Maybe I remember something wrong.
Does accessor already have storing value in double and using value as float?
or it only miss benchmark there

Reply:
No, we did not benchmark this. But I think in case there were some performance issues, this should be fixed in the accessor :)